### PR TITLE
resolve #5842 add deprecation warnings for 'conda env upload' and 'conda env attach'

### LINKED ISSUE #5842: Deprecate 'conda env attach' and 'conda env upload'
These two commands are outside the scope and spirit of conda.  The first requires knowledge of a "notebook," which is not one of conda's fundamental units of work.  The second requires knowledge of the Anaconda API, which is the purview of anaconda-client, not conda.

Deprecation warnings will be given in conda 4.4, with the command scheduled for removal in conda 4.5.

### PR DESCRIPTION
**Pull request by kalefranz**
resolves #5842

### CODE REVIEW

File: conda_env/specs/binstar.py [hunk 1]
```diff
@@ -39,6 +39,14 @@ def __init__(self, name=None, **kwargs):
             self.binstar = None
 
     def can_handle(self):
+        result = self._can_handle()
+        if result:
+            print("WARNING: Binstar environments are deprecated and scheduled to be "
```
Comment (goanpeca) [inline conda_env/specs/binstar.py]: Binstar?

Comment (kalefranz) [inline conda_env/specs/binstar.py]: It shows up elsewhere in user messages in this code too.  I actually spent time figuring out how this code path would even be triggered, and I don't think it's in use anywhere.  I don't think there is anything on the internet documented on how you even use it.  So I don't think this should be much of an issue.

**Comment by kalefranz** [comment]
If you're happy with the ^^^ explanation, let's merge :)

**Comment by mforbes** [comment]
@kalefranz 
So what are users supposed to do?  I got this message while trying to update an environment from Anaconda cloud as described here:

* http://docs.anaconda.com/anaconda-cloud/user-guide/tasks/work-with-environments/

```bash
$ conda env create mforbes/jupyter
WARNING: Binstar environments are deprecated and scheduled to be removed in conda 4.5. See conda issue #5843 at https://github.com/conda/conda/pull/5843 for more information.
```

I am not aware of using Binstar (though might have long ago), and the documentation says nothing about this deprecation.

**Pull request merged**